src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -81,16 +81,21 @@ public ObjectAdapter(IContractResolver contractResolver)
         /// </summary>
         /// <param name="operation">The add operation</param>
         /// <param name="objectApplyTo">Object to apply the operation to</param>
-        public void Add(Operation<T> operation, T objectToApplyTo)
+        public void Add(Operation<T> operation, T objectToApplyTo, Action<string> action = null)
         {
-            Add(operation.path, operation.value, objectToApplyTo, operation);
+            Add(operation.path, operation.value, objectToApplyTo, operation, action);
         }
 
         /// <summary>
         /// Add is used by various operations (eg: add, copy, ...), yet through different operations;
         /// This method allows code reuse yet reporting the correct operation on error
         /// </summary>
-        private void Add(string path, object value, T objectToApplyTo, Operation<T> operationToReport)
+        private void Add(
+            string path,
+            object value,
+            T objectToApplyTo,
+            Operation<T> operationToReport,
+            Action<string> action = null)
```

Comment:
why default? That says to me that there's a bunch of code calling this without an error action.